salt/modules/ddns.py

```diff
@@ -7,13 +7,13 @@
     are a couple of optional configuration parameters made available to
     support this (the keyname is only needed if the keyring contains more
     than one key)::
-
-        keyfile: keyring file (default=None)
-        keyname: key name in file (default=None)
-        keyalgorithm: algorithm used to create the key
-                      (default='HMAC-MD5.SIG-ALG.REG.INT').
-            Other possible values: hmac-sha1, hmac-sha224, hmac-sha256,
-                hmac-sha384, hmac-sha512
+        ddns:
```

Comment:
```suggestion
        ddns.keyfile: keyring file (default=None)
```

Is there a reason not to just change the docs to add `ddns` before each one?  Seems this was the initial intent. And if we change it to `ddns:` instead then it would need to be properly deprecated, because any user using this module would need to update their configuration.

Reply:
The reason I have opted to do nested dict, is that I have not seen any module except this one use the `module_name.key: value` instead of `module_name: {key1: value}`.

I can make it supported both ways, and send a deprecation message about the "old" dot style, if you are fine with that?

Reply:
there are a lot of other modules that use this same format:

https://github.com/saltstack/salt/blob/v2019.2.0/salt/states/boto_lc.py#L34
https://github.com/saltstack/salt/blob/v2019.2.0/salt/states/boto_s3.py#L24
https://github.com/saltstack/salt/blob/v2019.2.0/salt/modules/redismod.py#L12

just to name a few. Do you see any other reason, since this is indeed used throughout salt? 

If there is, this is how you properly deprecate: https://docs.saltstack.com/en/latest/topics/development/deprecations.html